inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/gather.cpp

```diff
@@ -43,11 +98,31 @@ const auto params = testing::Combine(
         testing::Values(CommonTestUtils::DEVICE_CPU)
 );
 
-INSTANTIATE_TEST_CASE_P(
-        smoke_Gather,
-        GatherLayerTest,
-        params,
-        GatherLayerTest::getTestCaseName
+INSTANTIATE_TEST_CASE_P(smoke_Gather7_BD1, Gather7LayerTest, gather7ParamsSubset_BD1, Gather7LayerTest::getTestCaseName);
+
+const std::vector<std::vector<size_t>> indicesShapes_BD2 = {
+        std::vector<size_t>{4, 5, 4, 3},
+        std::vector<size_t>{4, 5, 3, 2}
+};
+
+const std::vector<std::tuple<int, int>> axes_BD2 = {
+        std::tuple<int, int>{2, 2},
+        std::tuple<int, int>{3, -2},
+        std::tuple<int, int>{-1, 2},
+};
+
+const auto gather7ParamsSubset_BD2 = testing::Combine(
+        testing::ValuesIn(inputShapes4D),
+        testing::ValuesIn(indicesShapes_BD2),
+        testing::ValuesIn(axes_BD2),
+        testing::ValuesIn(netPrecisions),
+        testing::Values(InferenceEngine::Precision::UNSPECIFIED),
+        testing::Values(InferenceEngine::Precision::UNSPECIFIED),
+        testing::Values(InferenceEngine::Layout::ANY),
+        testing::Values(InferenceEngine::Layout::ANY),
+        testing::Values(CommonTestUtils::DEVICE_CPU)
 );
 
+INSTANTIATE_TEST_CASE_P(smoke_Gather7_BD2, Gather7LayerTest, gather7ParamsSubset_BD2, Gather7LayerTest::getTestCaseName);
+
 }  // namespace
```

Comment:
Could you add test for negative indices?

Reply:
Now Gather-7 isn't supported negative indexes
However I added more tests with negative axes and batch_dims

Reply:
Implementation processes negative indices and it should be validated.

Reply:
Reference impl has another [behavior](https://github.com/openvinotoolkit/openvino/blob/115e9f5d6db629b28df7aa1e2721611e7da5f4d7/ngraph/core/reference/include/ngraph/runtime/reference/gather.hpp#L56) for negative indexes
In CPU plugin we have impl which doesn't drop accuracy unlike the reference behavior

So the tests will fail